resources/js/components/Pagination.vue

```diff
@@ -0,0 +1,114 @@
+                </svg>
+            </button>
+        </li>
+
+        <li v-for="page in pages" :key="page.name" class="py-2 px-3 leading-tight text-black bg-white-700 border border-gray-300 hover:bg-cyan-600 hover:text-white">
+            <button type="button" @click="onClickPage(page.name)" :disabled="page.isDisabled">
+                {{ page.name }}
+            </button>
+        </li>
+
+        <li class="py-2 px-3 leading-tight text-black bg-white-700 border border-gray-300 hover:bg-cyan-600 hover:text-white">
+            <button type="button" @click="onClickNextPage" :disabled="isInLastPage">
+                <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="w-3 h-3">
+                    <path stroke-linecap="round" stroke-linejoin="round" d="M8.25 4.5l7.5 7.5-7.5 7.5" />
+                </svg>
+            </button>
+        </li>
+
+        <li class="py-2 px-3 leading-tight text-black bg-white-700 rounded-r-lg border border-gray-300 hover:bg-cyan-600 hover:text-white">
+            <button type="button" @click="onClickLastPage" :disabled="isInLastPage">
+                <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="w-3 h-3">
+                    <path stroke-linecap="round" stroke-linejoin="round" d="M11.25 4.5l7.5 7.5-7.5 7.5m-6-15l7.5 7.5-7.5 7.5" />
+                </svg>
+            </button>
+        </li>
+    </ul>
+</template>
+
+<script>
+export default {
+    props: {
+        maxVisibleButtons: {
+            type: Number,
+            required: false,
+            default: 3
+        },
+        totalPages: {
+            type: Number,
+            required: true
+        },
+        perPage: {
+            type: Number,
+            required: true
+        },
+        currentPage: {
+            type: Number,
+            required: true
+        }
+    },
+
+    computed: {
+        startPage() {
+            if (this.currentPage === 1) {
+                return 1
+            }
+
+            if (this.currentPage === this.totalPages) {
+                return this.totalPages - this.maxVisibleButtons
+            }
+
+            return this.currentPage - 1
+        },
+        isInFirstPage() {
+            return this.currentPage === 1
+        },
+        isInLastPage() {
+            return this.currentPage === this.totalPages
+        },
+        pages() {
+            const range = []
+
+            for (let i = this.startPage; i <= Math.min(this.startPage + this.maxVisibleButtons - 1, this.totalPages); i++) {
+                range.push({
+                    name: i,
+                    isDisabled: i === this.currentPage
+                })
+            }
+
+            return range
+        }
+    },
+    methods: {
+        onClickFirstPage() {
+            this.$emit('pagechanged', 1)
```

Comment:
"pagechanged" two different words.
Vue custom events(emits) convention is to use something like this `page:change`
Here, you'll be defining the the related thing (page) and the action you're firing (change)

This change should he applied for all the custom events in this component